leshan-core/src/main/java/org/eclipse/leshan/core/LwM2m.java

```diff
@@ -157,8 +157,8 @@ public static String validate(String version) {
         public int hashCode() {
             final int prime = 31;
             int result = 1;
-            result = prime * result + ((major == null) ? 0 : major.hashCode());
-            result = prime * result + ((minor == null) ? 0 : minor.hashCode());
+            result = prime * result + (int) major;
+            result = prime * result + (int) minor;
```

Comment:
When I generate `hashcode()` with **eclipse**. 
There is no cast here. 

Code looks like : 
```java
result = prime * result + major;
result = prime * result + minor;
```

Any reason to use cast here ?

Reply:
No specific reason, this was generated by IntelliJ :see_no_evil::slightly_smiling_face: